openwisp_notifications/tests/test_notifications.py

```diff
@@ -813,6 +813,23 @@ def test_delete_obsolete_tasks(self, mocked_task, *args):
         user.delete()
         mocked_task.assert_not_called()
 
+    def test_email_notif_without_notif_setting(self):
+        target_obj = self._get_org_user()
+        data = dict(
+            email_subject='Test Email subject', url='https://localhost:8000/admin'
+        )
+        Notification.objects.create(
+            actor=self.admin,
+            recipient=self.admin,
+            description='Test Notification Description',
+            verb='Test Notification',
+            action_object=target_obj,
+            target=target_obj,
+            data=data,
+            type="default",
+        )
+        self.assertEqual(len(mail.outbox), 1)
+
```

Comment:
@aagman945 this test does not trigger the bug. The bug only appears if notification setting is not present for the  user. 

In the end of the test, you are assert for presence of one email. But we don't want an email to be sent in this scenario. This depicts that the test does not tests the exception handling that  you just added. 

I would change this test as following:

```suggestion
def test_email_notification_without_notification_setting(self):
        target_obj = self._get_org_user()
        data = dict(
            email_subject='Test Email subject', url='https://localhost:8000/admin'
        )
        # Delete notification settings of the recipient
        self.admin.notificationsetting_set.all().delete()
        Notification.objects.create(
            actor=self.admin,
            recipient=self.admin,
            description='Test Notification Description',
            verb='Test Notification',
            action_object=target_obj,
            target=target_obj,
            data=data,
            type="default",
        )
        self.assertEqual(len(mail.outbox), 0)
```